### ISSUE TITLE
RET507 false positive only inside a function

### ISSUE THREAD

**Issue by trag1c** [charliermarsh/ruff#3285]
<table>
  <tr>
    <td>Works fine here:</td>
    <td>RET507 detected here:</td>
  </tr>
  <tr>
    <td>

```py
x = 5
for char in "abc":
    if char == "a":
        x = 1
    elif char == "b":
        x = 2
        continue
    else:
        x = 3
print(x)
```

</td>
<td>

```py
def foo() -> None:
    x = 5
    for char in "abc":
        if char == "a":
            x = 1
        elif char == "b":
            x = 2
            continue
        else:
            x = 3
    print(x)
```

</td>
</tr>
</table>

https://user-images.githubusercontent.com/77130613/222003075-0083d3c5-1996-401d-bdd0-77245c20d446.mov

**Comment by trag1c** [comment]
Actually, it looks like it could be a https://github.com/charliermarsh/ruff-vscode issue? I tried running `ruff --select RET507 file.py` on the very same code (on both `0.0.247` and `0.0.252`) and had no errors show up.

**Comment by charliermarsh** [comment]
Oh, this might've been fixed a while ago.

**Comment by charliermarsh** [comment]
I think this was fixed in #2881.

**Comment by charliermarsh** [comment]
Do you know what version of the VS Code extension you're on?

**Comment by charliermarsh** [comment]
Just confirmed that this is no longer an issue on `main`.
